File: src/main/java/com/project/pantau/dto/report/ReportResponse.java
package com.project.pantau.dto.report;

import com.project.pantau.dto.category.CategoryResponse;
import com.project.pantau.enums.ReportStatus;
import lombok.Builder;

import java.time.OffsetDateTime;
import java.util.UUID;

@Builder
public record ReportResponse(
        UUID id,
        CategoryResponse category,
        String description,
        String photoUrl,
        Double latitude,
        Double longitude,
        ReportStatus status,
        OffsetDateTime createdAt,
        OffsetDateTime updatedAt
) {
}

File: backend/internal/dto/report/report_response.go
package report

import (
	"github.com/google/uuid"
	"pantau/internal/dto/category"
	"pantau/internal/enums"
	"time"
)

type ReportResponse struct {
	ID          uuid.UUID                 `json:"id"`
	Category    category.CategoryResponse `json:"category"`
	Description *string                   `json:"description"`
	PhotoURLs   []string                  `json:"photo_urls"`
	Latitude    float64                   `json:"latitude"`
	Longitude   float64                   `json:"longitude"`
	Status      enums.ReportStatus        `json:"status"`
	CreatedAt   time.Time                 `json:"created_at"`
	UpdatedAt   time.Time                 `json:"updated_at"`
}
